exporter/signalfxexporter/dpclient.go

```diff
@@ -77,56 +74,32 @@ type sfxDPClient struct {
 func (s *sfxDPClient) pushMetricsData(
 	ctx context.Context,
 	md pdata.Metrics,
-) (droppedTimeSeries int, err error) {
-	metricsData := internaldata.MetricsToOC(md)
-	var numDroppedTimeseries int
+) (droppedDataPoints int, err error) {
+	var numDroppedDataPoints int
 	var errs []error
 
-	var currentToken string
-	var batchStartIdx int
-	for i := range metricsData {
-		metricToken := s.retrieveAccessToken(metricsData[i])
-		if currentToken != metricToken {
-			if batchStartIdx < i {
-				// TODO:
-				// 1) To leverage the queued retry mechanism available in new exporters,
-				// 	  look for error cases that are non-permanent and track the list of metric
-				//    data that needs to be retried. This will be done once partial retry support
-				//    is added for metrics type.
-				// 2) Also consider invoking in a goroutine to some concurrency going, in case
-				//    there are a lot of tokens.
-				droppedCount, err := s.pushMetricsDataForToken(ctx, metricsData[batchStartIdx:i], currentToken)
-				numDroppedTimeseries += droppedCount
-				if err != nil {
-					errs = append(errs, err)
-				}
-				batchStartIdx = i
-			}
-			currentToken = metricToken
-		}
-	}
-
-	// Ensure to get the last chunk of metrics.
-	if len(metricsData[batchStartIdx:]) > 0 {
-		droppedCount, err := s.pushMetricsDataForToken(ctx, metricsData[batchStartIdx:], currentToken)
-		numDroppedTimeseries += droppedCount
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
```

Comment:
Need to check if rms.At(i) is nil and skip that. Same everywhere you iterate over arrays.

Reply:
Done.